### PR TITLE
Add uninstall script

### LINKED ISSUE #29: Add uninstall script
It would be immensely useful to have a counterpart to `setupShellInitFile()`, something like `cleanupShellInitFile()`.

That would allow to run `setupShellInitFile()` in an npm `postinstall` hook. Doing this right now will append another completion block to the shell init file each time `npm i -g cli-package` is run (e.g. to update the CLI).

`cleanupShellInitFile()` could be used in a `postuninstall` hook to make sure there will never be more than one completion block.

### PR DESCRIPTION
**Pull request by Loilo**
[As promised](https://github.com/f/omelette/issues/29#issuecomment-424851045), here's a PR to add an uninstall script to omelette (resolves #29). I called it `cleanupShellInitFile()`.

### Refactoring

This change required some larger-scoped modifications to keep things DRY, as I had to pull out some logic from `setupShellInitFile()` for reusage in `cleanupShellInitFile()`. It looks like a lot, but it's mostly pushing existing lines of code around in the class.

### Change of existing behaviour

I introduced one (kind of breaking) change to omelette's existing behaviour (and would be glad to hear feedback on this):

Currently, omelette fails silently ([just goes `process.exit()`](https://github.com/f/omelette/blob/707e103d8fa9af2a7f17054cb5a7a8a72f8a2720/coffee/omelette.coffee#L187-L202)) when the user's shell is not supported. This makes it impossible for developers to recognize and react to that problem.

I've changed this to be more explicit: omelette now throws an error when trying to `setup/cleanupShellInitFile()` in unsupported shells.

> I realize that this change is far out of scope of #29. It came up during the mentioned refactoring, when I faced the possibility that `@shell` might be undefined either because `@getActiveShell()` was never called or because it actually returned `undefined`.
>
> This required a solution, and I concluded that throwing an Error would probably be superior to just returning nothing.

Since omelette is pre 1.0.0, this change is valid SemVer and its impact is rather low (because shells other than `bash/zsh/fish` should be pretty rare in UNIX environments).

Windows has no `process.env.SHELL` and thus omelette [has always thrown](https://github.com/f/omelette/blob/707e103d8fa9af2a7f17054cb5a7a8a72f8a2720/coffee/omelette.coffee#L160) there anyway, the error message just got a little clearer there (`Shell could not be detected` instead of `TypeError: Cannot read property 'match' of undefined`).

So in conclusion, the new behaviour doesn't really change a lot. Both, the silent `process.exit()` as well as an uncaught error, should in most cases end script execution. The error just creates additional error output.

Please review and consider this change carefully.

### Documentation

Usage of `cleanupShellInitFile()` is documented in the readme.

### Testing

I'd be happy if these changes were tested by others as well. I got it to work in my `zsh` environment, but neither do I have `fish` installed nor does my `bash` work properly (not due to these changes, but in general).

### CODE REVIEW

**Comment by f** [comment]
Thank you for this great PR. I'm going to test it on my Fish and Bash and will approve. The detection exception is a good idea. Brilliant.
